src/security/Io.cc

```diff
@@ -83,10 +83,16 @@ Security::Handshake(Comm::Connection &transport, const ErrorCode topError, Fun i
         ; // fall through to handle the problem
     }
 
+    if (const auto verifyErrorDetailRaw = SSL_get_ex_data(connection, ssl_ex_index_ssl_error_detail)) {
+        const auto verifyErrorDetail = *static_cast<ErrorDetail::Pointer*>(verifyErrorDetailRaw);
+        return IoResult(verifyErrorDetail);
+    }
```

Comment:
> We could probably successfully argue that whoever built the original errorDetail object (extracted here) should have setPeerCertificate() already

Yes it have it. I carefully check it.

> However, we clearly cannot say the same about the ioResult.errorDescription and, possibly, ioResult.important fields.

commit d7898cfefa
Maybe in the case of validation error should add "validation error" as description.

Reply:
> Maybe in the case of validation error should add "validation error" as description.

Maybe, but I am worried about relying too much on knowing what ssl_ex_index_ssl_error_detail contains. You assume that it is a validation error, and maybe that is all it can be today, but the code cannot really know that. I will post new change requests in hope to address that concern.

I consider this change request addressed, thank you.